packages/metro-react-native-babel-transformer/src/index.js

```diff
@@ -161,7 +166,7 @@ function buildBabelConfig(filename, options, plugins?: BabelPlugins = []) {
   return Object.assign({}, babelRC, config);
 }
 
-function transform({filename, options, src, plugins}: BabelTransformerArgs) {
+async function transform({filename, options, src, plugins}: BabelTransformerArgs) {
```

Comment:
What about the callsite of this? Does it already `await` or work with Promises?

Reply:
Hi @cpojer! Thanks for taking a look. The answer is yes, it does!

I put together a demo repo here that uses `patch-package` to apply these changes.

https://github.com/jrylan/metro-react-native-babel-transformer-async-demo/tree/master

I've added notes in the repo README that shows that it works.